libbeat/autodiscover/providers/jolokia/jolokia.go

```diff
@@ -19,15 +19,16 @@ package jolokia
 
 import (
 	"github.com/elastic/beats/libbeat/autodiscover"
+	"github.com/elastic/beats/libbeat/autodiscover/providers"
 	"github.com/elastic/beats/libbeat/autodiscover/template"
 	"github.com/elastic/beats/libbeat/common"
 	"github.com/elastic/beats/libbeat/common/bus"
 	"github.com/elastic/beats/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/libbeat/feature"
 )
 
-func init() {
-	autodiscover.Registry.AddProvider("jolokia", AutodiscoverBuilder)
-}
+// Feature exposes the Jolokia autodiscovery provider.
+var Feature = providers.Feature("jolokia", AutodiscoverBuilder, feature.Experimental)
```

Comment:
The autodiscover builder logs an experimental warning. If possible the warnings about beta/experimental features should be logged by the registry. 

How about adding support for 'Name' and 'doc strings' to features.  In the future we will also have to add a config schema here. We might even register a fields.yml here, for building the actual event schema of a running beat. Some of these might be optional.
This allows us to create commands to print available features. This makes me wonder if we want to introduce a generic such that a Feature is registered like this:

```
var Feature = providers.Feature("jolokia", AutodiscoverBuilder, feature.Details{
  FullName: "Jolokia Discovery",
  Doc: "<short one liner doc>",
  Stability: feature.Experimental,
  Settings: ... // <schema definition>,
  Fields: ... // <fields.yml> documenting fields being added to events
  FullDoc: ... // very long doc string (or function for producing doc strings... have developer create docs in "docs.go")
})
```

With these information we can:
- build config validation
- create a minimal fields.yml
- provide a command `filebeat features all`
- provide a command like `filebeat help autodiscovery.jolokia` to print module docs

Reply:
If things go that route central management could use this too, so we can reload a full feature from a new config.

Reply:
@dedemorton We can still keep all the docs outside the main go code and just provide helpers method that will read the files when we create the `Feature.Detail`.

@urso I will make the changes to receive a `feature.Detail` instead of a stability for future proofing it.